script.sh

```diff
@@ -177,12 +178,28 @@ fi
 
 cat >>Dockerfile <<EOF
 RUN apt-get update && apt-get dist-upgrade --yes
-RUN apt-get install --yes --no-install-recommends build-essential equivs devscripts git-buildpackage
+RUN apt-get install --yes --no-install-recommends build-essential equivs devscripts git-buildpackage apt-transport-https wget ca-certificates
 
 WORKDIR $(pwd)
 COPY . .
 EOF
 
+if [ -n "${TRAVIS_EXTRA_REPOSITORY_GPG_URL}" ]
+then
+	cat >>Dockerfile <<EOF
+RUN wget -O- "${TRAVIS_EXTRA_REPOSITORY_GPG_URL}" | apt-key add -
+EOF
+fi
+
+if [ -n "${TRAVIS_EXTRA_REPOSITORY}" ]
+then
+	cat >>Dockerfile <<EOF
+RUN echo "deb ${TRAVIS_EXTRA_REPOSITORY}" >> /etc/apt/sources.list
+RUN echo "deb-src ${TRAVIS_EXTRA_REPOSITORY}" >> /etc/apt/sources.list
+RUN apt-get update
```

Comment:
I can't find any later invocation of `apt-get update`, only the one above on line 180. I'm doing it here again, because the extra repository may use HTTPS, in which case I need `apt-transport-https` installed before I attempt to download from it. The process goes as follows:

1. add standard Debian repositories
2. `apt-get update && apt-get dist-upgrade`
3. install some base tools, `apt-transport-https` among them
4. add extra repository
5. `apt-get update` again in order to get package list from extra repository

That's also why I see the change in `apt-get install` you commented on above as an integral part of this commit and would rather if we kept everything in one piece... unless you really want me to split it.